src/core/client/stream/tabs/comments/containers/PostCommentFormContainer.tsx

```diff
@@ -60,6 +69,9 @@ export class PostCommentFormContainer extends Component<Props, State> {
       form.reset({});
     } catch (error) {
       if (error instanceof InvalidRequestError) {
+        if (shouldTriggerSettingsRefresh(error.code)) {
+          await this.props.refreshSettings();
+        }
```

Comment:
This pattern appears more than once, is it possible to extract this?

Reply:
For a good abstraction it would require a Higher Order Component which adds other complexities, not sure if there is a win here.